language/libra-vm/src/libra_vm.rs

```diff
@@ -336,16 +336,18 @@ impl LibraVM {
         let mut failed_gas_left = GasUnits::new(0);
         match payload {
             VerifiedTransactionPayload::Module(m) => {
-                self.move_vm.publish_module(m, &mut data_store, txn_data)
+                self.move_vm
+                    .publish_module(m, txn_data.sender(), &mut data_store)
             }
             VerifiedTransactionPayload::Script(s, ty_args, args) => {
                 let ret = self.move_vm.execute_script(
                     s,
                     ty_args,
                     args,
-                    &mut cost_strategy,
+                    txn_data.sender(),
+                    txn_data.transaction_size(),
                     &mut data_store,
-                    txn_data,
+                    &mut cost_strategy,
```

Comment:
Removing the idea of "transaction" from the interpreter it awesome!

I was wondering your thoughts on instead of passing the transaction size to the entrypoint, changing the `CostStrategy` to hold this?

To explain a bit more, `CostStrategy::transaction(...)` would take the `TransactionMetadata` and would have an additional field `min_gas_charge` in it that is calculated from the transaction size. The `CostStrategy::system(...)` would just set this to `0`.

Then you pass only the `CostStrategy` to the `Interpreter::entrypoint` and `charge_intrinsic_gas` would no longer take the transaction size, but use this stored `min_gas_charge` within it.

Reply:
yes that is in line with the talk we had and I'd like to do it. But I was thinking in another PR so I can possibly come up with alternative solutions and we can see how we like it.
Thoughts?

Reply:
ah! For some reason I was thinking this was that PR....

Carry on! We can take a look at it in the upcoming PR